src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/UnconfiguredProjectContextProvider.cs

```diff
@@ -149,24 +150,43 @@ private ProjectData GetProjectData()
             string languageName = await GetLanguageServiceName().ConfigureAwait(false);
             if (string.IsNullOrEmpty(languageName))
                 return null;
-            
+
             Guid projectGuid = await GetProjectGuidAsync().ConfigureAwait(false);
             string targetPath = await GetTargetPathAsync().ConfigureAwait(false);
 
             // Don't initialize until the project has been loaded into the IDE and available in Solution Explorer
             await _asyncLoadDashboard.ProjectLoadedInHost.ConfigureAwait(false);
 
             // TODO: https://github.com/dotnet/roslyn-project-system/issues/353
-            return await _taskScheduler.RunAsync(TaskSchedulerPriority.UIThreadBackgroundPriority, async () => 
+            return await _taskScheduler.RunAsync(TaskSchedulerPriority.UIThreadBackgroundPriority, async () =>
             {
                 await _commonServices.ThreadingService.SwitchToUIThread();
 
                 var projectData = GetProjectData();
 
+                // LanguageServices requires a normalized absolute target path.
+                targetPath = NormalizeTargetPath(targetPath, projectData);
+
                 return _contextFactory.Value.CreateProjectContext(languageName, projectData.DisplayName, projectData.FullPath, projectGuid, _commonServices.Project.Services.HostObject, targetPath);
             });
         }
 
+        private static string NormalizeTargetPath(string targetPath, ProjectData projectData)
+        {
+            if (string.IsNullOrEmpty(targetPath))
+            {
+                targetPath = projectData.DisplayName;
+            }
+
+            if (!Path.IsPathRooted(targetPath))
+            {
+                var directory = Path.GetDirectoryName(projectData.FullPath);
+                targetPath = Path.Combine(directory, targetPath);
+            }
+
+            return targetPath;
```

Comment:
I don't see how this is correct - how is this data have anything to do with target path?